src/flatcollections-array/FlatArray/FlatArray.T/Functional/Functional.Chunk.cs

```diff
@@ -0,0 +1,40 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<FlatArray<T>> Chunk(int size)
+    {
+        if (size < 1)
+        {
+            throw InnerExceptionFactory.SizeOutsideBounds(nameof(size), size);
+        }
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var chunksLength = length / size;
+        if (length % size != default)
+        {
+            chunksLength++;
+        }
+
+        var chunks = new FlatArray<T>[chunksLength];
+        for (var i = 0; i < chunks.Length; i++)
+        {
+            var start = i * size;

```

Comment:
Consider avoiding multiplication in each iteration, in a way like following.

Note: `unchecked` is used when incrementing the `start` to emphasize overflow may occur for huge arrays (and this overflowed value will not affect since it may occur only in the end of the last iteration and further will not be used).

```cs
int start = 0;
for (...)
{
    // main work here
    // ...
    unchecked  { start += size; }
}
```